### ISSUE TITLE
Double invocation of lambda through SNS

### ISSUE THREAD

**Issue by rushikeshsane95** [aws/aws-sdk-js#3772]
Confirm by changing [x] to [x] below to ensure that it's a bug:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  
**Describe the bug**
I have a lambda function (L1) which publishes a message to SNS using javascript SDK **callback** function.
Another lambda function (L2) is subscribed to SNS.
When L1 publishes message to SNS, L2 is triggered twice. 
This happens only when L1 uses callback function as follows:

`await sns.publish(params, (err, data) => {
        if (err) console.log(err, err.stack); // an error occurred
        else console.log(`Message published to SNS: ${data}`); // successful response
      }).promise();`

Lambda L2 is invoked only once (which is desired) **without a callback function.**

`await sns.publish(params).promise();`

**Is the issue in the browser/Node.js?**
Node.js

**If on Node.js, are you running this on AWS Lambda?**
Yes

**Details of the browser/Node.js version**
v12.18.1

**SDK version number**
2.900.0

**To Reproduce (observed behavior)**
Code shared above.

**Expected behavior**
Lambda L2 should be invoked only once.

**Screenshots**
![Screen Shot 2021-05-19 at 14 26 43](https://user-images.githubusercontent.com/53753875/118812490-77489d80-b8ae-11eb-998b-628728895254.png)

**Comment by ajredniwja** [comment]
@rushikeshsane95 thanks for opening this issue, duplicate of https://github.com/aws/aws-sdk-js/issues/3347

Please check the above mentioned issue, and let us know if that doesnt help. 

Closing this issue now, please reach out if you have any other questions.
